monitoring/perf_flags.cc

```diff
@@ -0,0 +1,300 @@
+#include "monitoring/perf_flags_imp.h"
+namespace rocksdb {
+const PerfFlags PerfFlagsEnableCount = {
+    .perf_level = 2,
+    .enable_user_key_comparison_count_bit = 1,
+    .enable_block_cache_hit_count_bit = 1,
+    .enable_block_read_count_bit = 1,
+    .enable_block_read_byte_bit = 1,
+    .enable_block_cache_index_hit_count_bit = 1,
+    .enable_index_block_read_count_bit = 1,
+    .enable_block_cache_filter_hit_count_bit = 1,
+    .enable_filter_block_read_count_bit = 1,
+    .enable_compression_dict_block_read_count_bit = 1,
+    .enable_get_read_bytes_bit = 1,
+    .enable_multiget_read_bytes_bit = 1,
+    .enable_iter_read_bytes_bit = 1,
+    .enable_internal_key_skipped_count_bit = 1,
+    .enable_internal_delete_skipped_count_bit = 1,
+    .enable_internal_recent_skipped_count_bit = 1,
+    .enable_internal_merge_count_bit = 1,
+    .enable_get_from_memtable_count_bit = 1,
+    .enable_seek_on_memtable_count_bit = 1,
+    .enable_next_on_memtable_count_bit = 1,
+    .enable_prev_on_memtable_count_bit = 1,
+    .enable_seek_child_seek_count_bit = 1,
+    .enable_bloom_memtable_hit_count_bit = 1,
+    .enable_bloom_memtable_miss_count_bit = 1,
+    .enable_bloom_sst_hit_count_bit = 1,
+    .enable_bloom_sst_miss_count_bit = 1,
+    .enable_key_lock_wait_count_bit = 1,
+};
+
+const PerfFlags PerfFlagsEnableTimeExceptForMutex = {
+    .perf_level = 3,
+    .enable_user_key_comparison_count_bit = 1,
+    .enable_block_cache_hit_count_bit = 1,
+    .enable_block_read_count_bit = 1,
+    .enable_block_read_byte_bit = 1,
+    .enable_block_cache_index_hit_count_bit = 1,
+    .enable_index_block_read_count_bit = 1,
+    .enable_block_cache_filter_hit_count_bit = 1,
+    .enable_filter_block_read_count_bit = 1,
+    .enable_compression_dict_block_read_count_bit = 1,
+    .enable_get_read_bytes_bit = 1,
+    .enable_multiget_read_bytes_bit = 1,
+    .enable_iter_read_bytes_bit = 1,
+    .enable_internal_key_skipped_count_bit = 1,
+    .enable_internal_delete_skipped_count_bit = 1,
+    .enable_internal_recent_skipped_count_bit = 1,
+    .enable_internal_merge_count_bit = 1,
+    .enable_get_from_memtable_count_bit = 1,
+    .enable_seek_on_memtable_count_bit = 1,
+    .enable_next_on_memtable_count_bit = 1,
+    .enable_prev_on_memtable_count_bit = 1,
+    .enable_seek_child_seek_count_bit = 1,
+    .enable_bloom_memtable_hit_count_bit = 1,
+    .enable_bloom_memtable_miss_count_bit = 1,
+    .enable_bloom_sst_hit_count_bit = 1,
+    .enable_bloom_sst_miss_count_bit = 1,
+    .enable_key_lock_wait_count_bit = 1,
```

Comment:
Is it possible to deduplicate? Otherwise adding a new perf fields requires changing a lot of places.

Reply:
aha! macro could help